java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.nodes;
+
+import static io.fabric8.java.generator.nodes.Keywords.ADDITIONAL_PROPERTIES;
+import static io.fabric8.java.generator.nodes.Keywords.JAVA_UTIL_MAP;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.EnumDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import io.fabric8.java.generator.exceptions.JavaGeneratorException;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JObject extends AbstractJSONSchema2Pojo {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JObject.class);
+    private static final Set<String> IGNORED_FIELDS = new HashSet<>();
+
+    static {
+        IGNORED_FIELDS.add("description");
+        IGNORED_FIELDS.add("schema");
+        IGNORED_FIELDS.add("example");
+        IGNORED_FIELDS.add("examples");
+    }
+
+    private final String type;
+    private final Map<String, AbstractJSONSchema2Pojo> fields;
+    private final Set<String> required;
+    private JObjectOptions options;
+
+    public JObject(
+            String type,
+            Map<String, JSONSchemaProps> fields,
+            List<String> required,
```

Comment:
Annotate? `@Nullable`

Reply:
we don't have a `Nullable` annotation on the `compile` classpath ATM.